deeptools/computeMatrix.py

```diff
@@ -0,0 +1,418 @@
+    ... -S ../deeptools/test/test_heatmapper/test.bw \
+    ... -b 100 -a 100 --outFileName /tmp/_test.mat.gz \
+    ... -bs 1 -p 1".split())
+    >>> main(args)
+    >>> os.system('gunzip -f /tmp/_test.mat.gz')
+    0
+    >>> filecmp.cmp('../deeptools/test/test_heatmapper/master.mat',
+    ... '/tmp/_test.mat')
+    True
+    >>> os.remove('/tmp/_test.mat')
+    >>> args = parseArguments("scale-regions \
+    ... -R ../deeptools/test/test_heatmapper/test2.bed \
+    ... -S ../deeptools/test/test_heatmapper/test.bw \
+    ... -b 100 -a 100 -m 100 --outFileName /tmp/_test2.mat.gz \
+    ... -bs 10 -p 1".split())
+    >>> main(args)
+    >>> os.system('gunzip -f /tmp/_test2.mat.gz')
+    0
+    >>> filecmp.cmp('../deeptools/test/test_heatmapper/master_scale_reg.mat',
+    ... '/tmp/_test2.mat')
+    True
+    >>> os.remove('/tmp/_test2.mat')
+    """
+
+    args = process_args(args)
+
+    # concatenate intermediary bedgraph files
+    bed_file = open(deeptools.utilities.getTempFileName(suffix='.bed'), 'w+t')
+    if args.verbose:
+        print "temporary bed file {} created".format(bed_file.name)
+
+    if len(args.regionsFileName) > 1:
+        for bed in args.regionsFileName:
+            bed.close()
+            # concatenate all intermediate tempfiles into one
+            print "appending {} file".format(bed.name)
+            shutil.copyfileobj(open(bed.name, 'U'), bed_file)
+            # append hash and label based on the file name
+            label = os.path.basename(bed.name)
+            if label.endswith(".bed"):
+                label = label[:-4]
+            bed_file.write("# {}\n".format(label))
+        bed_file.seek(0)
+    else:
+        bed_file = args.regionsFileName[0]
+
+    parameters = {'upstream': args.beforeRegionStartLength,
+                  'downstream': args.afterRegionStartLength,
+                  'body': args.regionBodyLength,
+                  'bin size': args.binSize,
+                  'ref point': args.referencePoint,
+                  'verbose': args.verbose,
+                  'bin avg type': args.averageTypeBins,
+                  'missing data as zero': args.missingDataAsZero,
+                  'min threshold': args.minThreshold,
+                  'max threshold': args.maxThreshold,
+                  'scale': args.scale,
+                  'skip zeros': args.skipZeros,
+                  'nan after end': args.nanAfterEnd,
+                  'proc number': args.numberOfProcessors,
+                  'sort regions': args.sortRegions,
+                  'sort using': args.sortUsing
+                  }
+
+    hm = heatmapper.heatmapper()
+
+    scores_file_list = [x.name for x in args.scoreFileName]
+    hm.computeMatrix(scores_file_list, bed_file, parameters, verbose=args.verbose)
+    if args.sortRegions != 'no':
+        hm.matrix.sort_groups(sort_using=args.sortUsing, sort_method=args.sortRegions)
+
+    hm.saveMatrix(args.outFileName)
+    bed_file.close()
+    #os.remove(bed_file.name)
```

Comment:
I just fixed that a while ago while addressing another issue. Thanks for pointing out.